library/ssl_ticket.c

```diff
@@ -66,6 +66,11 @@
 
 #include <string.h>
 
+/* Byte reading macros */
+#define BYTE_0( x ) ( (uint8_t) (   ( x )         & 0xff ) )
+#define BYTE_1( x ) ( (uint8_t) ( ( ( x ) >> 8  ) & 0xff ) )
+#define BYTE_2( x ) ( (uint8_t) ( ( ( x ) >> 16 ) & 0xFF ) )
```

Comment:
inconsistent capitalisation of 0xff and 0xFF. Capitalised hex digits seem more common in the codebase, although it's not very consistent.

Reply:
You got me curious because I was under the impression that lowercase hex digits are more common. Turns out that's heavily the case in test data, which we get to see more because we change it a lot, whereas constants in the source code tend to be write-and-forget:
```
$ cat tests/suites/*.data |egrep -o -i '"[0-9a-f]+"' |tr -dc A-F |wc -c
47328
$ cat tests/suites/*.data |egrep -o -i '"[0-9a-f]+"' |tr -dc a-f |wc -c
1091558
```
But uppercase is more common in C code:
```
$ git ls-files '*.c' '*.h' '*.function' |xargs cat | egrep -o -i '0x[0-9a-f]+' |tr -dc A-F |wc -c
27346
$ git ls-files '*.c' '*.h' '*.function' |xargs cat | egrep -o -i '0x[0-9a-f]+' |tr -dc a-f |wc -c
13754
```
Though a lot of the uppercase is in one specific file:
```
$ git ls-files '*.c' '*.h' '*.function' ':!library/ecp_curves.c' |xargs cat | egrep -o -i '0x[0-9a-f]+' |tr -dc A-F |wc -c
9662
$ git ls-files '*.c' '*.h' '*.function' ':!library/ecp_curves.c' |xargs cat | egrep -o -i '0x[0-9a-f]+' |tr -dc a-f |wc -c
13751
```

So there's no convention and it doesn't matter, but it is a bit more readable to avoid mixed case:
```
$ git grep -E -o '0[Xx][0-9]*([a-f][0-9a-f]*[A-F]|[A-F][0-9A-F]*[a-f])'
library/cmac.c:0xbF
tests/data_files/server5.ku-ds.crt:0x5Fe
```

Reply:
I am just going to change the mix in the same file, but what I found was that when casting to `uint8_t` lower case was being used, and when casting to `unsigned char` upper case was being used. I don't know if that was done on purpose or purely coincidental, but I was planning on maintaining that in the macro definitions

Reply:
This is not a technical choice. Probably a matter of personal preference for both, with a small sample size (most of the code was written by a small number of people) so the numbers for `uint8_t` vs `unsigned char` and for `0xff` vs `0xFF` don't reflect a statistical idealization.

Reply:
ah makes sense